Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -687,6 +687,10 @@ def installVolumeRenderActorPipeline(self):
         colors, opacity = self.getColorOpacityForVolumeRender()
 
         self.volume_property.SetColor(colors)
+
+        self.setDefaultScalarOpacity(self.volume_property.GetScalarOpacity())
+        self.setDefaultGradientOpacity(self.volume_property.GetGradientOpacity())
```

Comment:
The names of these methods are confusing:

1. they have `default` but require a parameter
2. they are called independently of the mode of volume opacity render

Also the code in the lines below seems a bit muddled up. I think what should happen is:

1. set data for mapper
2. define which opacity method to use, with methods like `setVolumeOpacityToScalar` and `setVolumeOpacityToGradient` or with `setVolumeRenderOpacityMethod` (this will need to check if `self.volume_render_initialised == True` before calling `updateVolumePipeline` or better put the check in the latter method)
3. select colormap (notice that this is handled by an instance member
4. create `color` and `opacity`
5. finish to install the pipeline

Reply:
It seems the way it works (or what I understand) is that there is always a 'gradient opacity' and a 'scalar opacity'. which is why before we set any opacities we have to save the 2 default values, so they can be used later when we change the method we use. I could change the name instead to 'original' instead of default?
regarding point 2:
 - `updateVolumePipeline` already has the check for `volume_render_initialised`. 
 - I don't understand why we would use `setVolumeRenderOpacity` method at this point, as this will have been set by the user, or if it hasn't been set then when we later call `self.getVolumeRenderOpacityMethod()` we get 'gradient'

Reply:
I'll check on the colormap code and make some additions there, I'll add changing the colormap to the example to help with that